internal/transport/handler_server.go

```diff
@@ -141,7 +141,10 @@ type serverHandlerTransport struct {
 	stats []stats.Handler
 }
 
-func (ht *serverHandlerTransport) Close() {
+func (ht *serverHandlerTransport) Close(err error) {
+	if logger.V(logLevel) {
+		logger.Infof("Closing serverHandlerTransport: %v", err)
+	}
```

Comment:
Should this go inside the once instead?  Or be rephrased to indicate that close was called but that the stream may already have been closed?